workitem/workitem_repository.go

```diff
@@ -95,6 +97,43 @@ func (r *GormWorkItemRepository) Load(ctx context.Context, ID string) (*app.Work
 	return convertWorkItemModelToApp(wiType, res)
 }
 
+// LoadTopWorkitem returns top most work item of the list. Top most workitem has the Highest order.
+// returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemRepository) LoadTopWorkitem(ctx context.Context) (*app.WorkItem, error) {
+	res := WorkItem{}
+	r.db = r.db.Debug()
+	tx := r.db.Order("execution_order desc").Last(&res)
+	if tx.RecordNotFound() {
+		return nil, nil
+	}
+	if tx.Error != nil {
+		return nil, errors.NewInternalError(tx.Error.Error())
+	}
+	wiType, err := r.witr.LoadTypeFromDB(ctx, res.Type)
+	if err != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	return convertWorkItemModelToApp(wiType, &res)
+}
+
+// LoadBottomWorkitem returns bottom work item of the list. Bottom most workitem has the lowest order.
+// returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemRepository) LoadBottomWorkitem(ctx context.Context) (*app.WorkItem, error) {
+	res := WorkItem{}
+	tx := r.db.Order("execution_order").Last(&res)
```

Comment:
Btw. you do `tx := r.db.Order("execution_order desc").Last(&res)` also in `LoadTopWorkitem`. Shouldn't one find the minimum and one find the maximum execution order value instead of both finding the maximum execution order?